numpy/core/tests/test_function_base.py

```diff
@@ -113,6 +113,33 @@ def test_basic(self):
         assert_array_equal(y, [-100, -10, -1])
         assert_array_equal(y.imag, 0)
 
+    def test_boundaries_match_start_and_stop_exactly(self):
+        # make sure that the boundaries of the returned array exactly
+        # equal 'start' and 'stop' - this isn't obvious because
+        # np.exp(np.log(x)) isn't necessarily exactly equal to x
+        start = 0.3
+        stop = 20.3
+
+        y = geomspace(start, stop, num=1)
+        assert_equal(y[0], start)
+
+        y = geomspace(start, stop, num=1, endpoint=False)
+        assert_equal(y[0], start)
+
+        y = geomspace(start, stop, num=3)
+        assert_equal(y[0], start)
+        assert_equal(y[-1], stop)
+
+        y = geomspace(start, stop, num=3, endpoint=False)
+        assert_equal(y[0], start)
+
+    def test_nan_midpoint(self):
+        with errstate(all='ignore'):
+            y = geomspace(-3, 3, num=3)
+            assert_equal(y[0], -3.0)
+            assert_(isnan(y[1]))
```

Comment:
Perhaps a clearer test as:
```suggestion
            y = geomspace(-3, 3, num=4)
            assert_equal(y[0], -3.0)
            assert_(isnan(y[1:-1]).all())
```
It's not the midpoint that's nan, it's everything but the endpoint

Reply:
also good, done